tests/unit_tests/pyanaconda_tests/modules/storage/test_module_device_tree.py

```diff
@@ -688,6 +688,12 @@ def test_unlock_device(self, device_setup, device_teardown, format_setup):
         self.storage.devicetree.populate.assert_called_once()
         self.storage.devicetree.teardown_all.assert_called_once()
         assert dev2.format.has_key
+        assert self.interface.GetFormatData("dev2") == {
```

Comment:
Would be great to have this check also before unlocking the device to check if `has_key` is `False`.

Reply:
So before unlocking the has_key does not exist. I am also a bit confused why it's not false, but since I just parse it from blivet and I don't get something explicit there, I guessed it's correct.

Reply:
updated the test to reflect this.

Reply:
I think that it is possible that `fmt.has_key` returns `None` instead of `False`, so we throw it away when we prune the collected attributes. It would be great if this attribute was always there for `luks` formats, but in that case, that should be fixed in Blivet. @vojtechtrefny?

Reply:
> It would be great if this attribute was always there for `luks` formats, but in that case, that should be fixed in Blivet.

Makes sense. https://github.com/storaged-project/blivet/pull/1142